### ISSUE TITLE
@fetch-messages and @fetch-more-rooms events not being emit 

### ISSUE THREAD

**Issue by mkornyev** [antoine92190/vue-advanced-chat#234]
## Describe the bug

When scrolling up in the chat, the `@fetch-messages` event is not being emitted at all -- apart from when the room is first opened. `@fetch-more-rooms` is not being emit at any point. 

## Steps to reproduce

1. Render the `<chat-window>` 
2. Attach debug/print functions to the `@fetch-messages` && `@fetch-more-rooms` events 
3. Watch the former get triggered only when a room is loaded, and not when the user scrolls to the top of the window. Watch the latter not get triggered at all.

## Expected behavior

The @fetch-messages event is emit right after scrolling & reaching the top of the chat window 
The @fetch-more-rooms event is emit right after scrolling & reaching the bottom of the rooms panel

## Device (please complete the following information)

- OS: macOS (Big Sur)
- Browser: Chrome 94 
- Package version: 1.3.0
- Vue ^2.6.10

## Additional context

Perhaps I am misunderstanding how to trigger the events. Either way, would love to have this fixed for a production use case :)

**Comment by mkornyev** [comment]
It looks like the same erroneous behavior is happening for the @fetch-more-rooms event, nothing is being emitted on scroll. Although -- this was only tested with 2/3 rooms in in the rooms panel.

**Comment by antoine92190** [comment]
Could you share your component snippet or a minimal example to reproduce?

**Comment by mkornyev** [comment]
Sure. Here's my basic component usage (slot templates omitted)

```
  <chat-window
      :current-user-id="constants.USER_ID"
      :rooms="chat.rooms"
      :messages="chat.messages"
      :messages-loaded="!chat.messagesLoading"
      :rooms-loaded="!chat.roomsLoading"

      @send-message="sendMessage"
      @fetch-messages="(context) => context.options.reset ? enterChannel(context.room.roomId) : loadChannelMessages(true)"

      :messageActions="chatMessageActions"
      :show-reaction-emojis="false"
      :show-new-messages-divider="false"
      :styles="chat.styles"

      @fetch-more-rooms="debug('FETCH MORE ROOMS')"

      ...some more events being caught & debugged here...
    >
      <template 
        v-slot:room-header="{ room, userStatus }"
      >
        ...
      </template>

      <template v-slot:microphone-icon>
            ...
      </template>
    </chat-window>
```

**Comment by antoine92190** [comment]
Can you try updating vue version to 2.6.14?
https://github.com/antoine92190/vue-advanced-chat/releases/tag/0.8.9

**Comment by antoine92190** [comment]
I should probably add this information in the README

**Comment by antoine92190** [comment]
Just added Vue min version in README https://github.com/antoine92190/vue-advanced-chat/commit/ff9f4a973e291cbd566d4f1e016f7fdfc9087c98

**Comment by antoine92190** [comment]
This should be fixed using vue version `2.6.14`.
Please reopen if that's not the case.

**Comment by omitobi** [comment]
Hello @antoine92190 I have similar issue raised by @mkornyev Even after updating my vue version to 2.6.14 `@fetch-more-rooms` event is not triggered at all.

Every other events work except `fetch-more-rooms`

**Comment by antoine92190** [comment]
Have you passed rooms-loaded prop?
It should be false until all rooms are loaded, otherwise fetch-more-room will not be triggered

**Comment by omitobi** [comment]
Indeed I passed `rooms-loaded` as `false` and as expected it shows a loading icon but the event does not get triggered.
<img width="459" alt="Screenshot 2022-01-05 at 11 42 33" src="https://user-images.githubusercontent.com/16482234/148196384-857403ba-ba7c-4d0f-8673-2a6227dffca1.png">

**Comment by omitobi** [comment]
I have added this codesandbox to show what I meant: https://codesandbox.io/s/interesting-cloud-gghib?file=/src/App.vue

**Comment by antoine92190** [comment]
I tested your sandbox, you need to make use of `loading-rooms` prop. It should be true until you uploaded the first batch of rooms

**Comment by omitobi** [comment]
You're right. With `loading-rooms` used that way paginating rooms work.

**Comment by NINI1988** [comment]
I had the issue that after each batch of rooms loaded the whole rooms list was recreated and scrolled up to the top.
I had misunderstood that `loading-room` is only allowed to be set until the first batch is loaded.
I have set `loading-room` before each api request to true and to false afterwards which caused this weird behaviour.

Maybe it should be more clear that `loading-room` is only allowed to be set until the first batch of rooms is loaded and should never be set again.

**Comment by jameswagoner** [comment]
@NINI1988 I experienced the same issue with the list scrolling to the top with setting `loading-room` to true then false with each pagination request.

Your mention of only setting the prop to false after the first page is loaded and then never setting to true again... are you still able to paginate as you scroll? I can only paginate when I set the prop back to true then false with each page request.

Just want to clarify your experience.
